conf/modules.config

```diff
@@ -0,0 +1,770 @@
+/*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    Config file for defining DSL2 per module options and publishing paths
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    Available keys to override module options:
+        ext.args   = Additional arguments appended to command in module.
+        ext.args2  = Second set of arguments appended to command in module (multi-tool modules).
+        ext.args3  = Third set of arguments appended to command in module (multi-tool modules).
+        ext.prefix = File name prefix for output files.
+----------------------------------------------------------------------------------------
+*/
+
+process {
+
+    publishDir = [
+        path: { "${params.outdir}/${task.process.tokenize(':')[-1].tokenize('_')[0].toLowerCase()}" },
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
+    ]
+
+    withName: SAMPLESHEET_CHECK {
+        publishDir = [
+            path: { "${params.outdir}/pipeline_info" },
+            mode: params.publish_dir_mode,
+            saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
+        ]
+    }
+
+    withName: FASTQC {
+        ext.args = '--quiet'
+    }
+
+    withName: CUSTOM_DUMPSOFTWAREVERSIONS {
+        publishDir = [
+            path: { "${params.outdir}/pipeline_info" },
+            mode: params.publish_dir_mode,
+            pattern: '*_versions.yml'
+        ]
+    }
+
+    // TRIMMING couresy of nf-core/rnaseq
+
+if (!params.skip_trimming) {
+    process {
+        withName: '.*:FASTQC_TRIMGALORE:TRIMGALORE' {
+            ext.args   = {
+                [
+                    "--fastqc_args '-t ${task.cpus}' ",
+                    params.trim_nextseq > 0 ? "--nextseq ${params.trim_nextseq}" : ''
+                ].join(' ').trim()
+            }
+            publishDir = [
+                [
+                    path: { "${params.outdir}/trimgalore/fastqc" },
+                    mode: params.publish_dir_mode,
+                    pattern: "*.{html,zip}"
+                ],
+                [
+                    path: { "${params.outdir}/trimgalore" },
+                    mode: params.publish_dir_mode,
+                    pattern: "*.fq.gz",
+                    enabled: params.save_trimmed
+                ],
+                [
+                    path: { "${params.outdir}/trimgalore" },
+                    mode: params.publish_dir_mode,
+                    pattern: "*.txt"
+                ]
+            ]
+        }
+    }
+}
+
+    // PREPARE GENOME
```

Comment:
I suspect you will get WARN messages if the indices already exist

Reply:
I'll test this. I have a hard time resolving WARN messages